wpcom/class-amp-polldaddy-embed.php

```diff
@@ -1,26 +1,100 @@
 <?php
+/**
+ * Class WPCOM_AMP_Polldaddy_Embed
+ *
+ * @package AMP
+ */
 
+/**
+ * Class WPCOM_AMP_Polldaddy_Embed
+ */
 class WPCOM_AMP_Polldaddy_Embed extends AMP_Base_Embed_Handler {
+
+	/**
+	 * Register embed.
+	 */
 	public function register_embed() {
 		add_shortcode( 'polldaddy', array( $this, 'shortcode' ) );
+		add_filter( 'embed_oembed_html', array( $this, 'filter_embed_oembed_html' ), 10, 3 );
 	}
 
+	/**
+	 * Unregister embed.
+	 */
 	public function unregister_embed() {
 		remove_shortcode( 'polldaddy' );
+		remove_filter( 'embed_oembed_html', array( $this, 'filter_embed_oembed_html' ), 10 );
 	}
 
+	/**
+	 * Shortcode.
+	 *
+	 * @param array $attr Shortcode attributes.
+	 * @return string Shortcode.
+	 */
 	public function shortcode( $attr ) {
+		global $wp_embed;
+
+		$output = '';
 		$url = 'https://polldaddy.com/';
 		if ( ! empty( $attr['poll'] ) ) {
 			$url .= 'poll/' . $attr['poll'] . '/';
-			$name = ! empty( $attr['title'] ) ? $attr['title'] : __( 'View Poll', 'amp' );
-		} elseif ( ! empty( $attr['survey'] ) ) { // Surveys and Quizzes both use attr survey
+		} elseif ( ! empty( $attr['survey'] ) ) {
 			$url .= 's/' . $attr['survey'] . '/';
-			$name = ! empty( $attr['title'] ) ? $attr['title'] : __( 'View Survey', 'amp' );
-		} else {
-			return ''; // We can't embed anything useful for rating
 		}
 
-		return '<p><a href="' . esc_url( $url ) . '">' . esc_html( $name ) . '</a></p>';
+		if ( ! empty( $attr['title'] ) ) {
+			$output = $this->render_link( $url, $attr['title'] );
+		} elseif ( $url ) {
+			$output = $wp_embed->shortcode( $attr, $url );
+		}
+
+		return $output;
+	}
+
+	/**
+	 * Filter oEmbed HTML for PollDaddy to for AMP output.
+	 *
+	 * @param string $cache Cache for oEmbed.
+	 * @param string $url   Embed URL.
+	 * @param array  $attr  Shortcode attributes.
+	 * @return string Embed.
+	 */
+	public function filter_embed_oembed_html( $cache, $url, $attr ) {
+		$parsed_url = wp_parse_url( $url );
+		if ( $url && false === strpos( $parsed_url['host'], 'polldaddy.com' ) ) {
```

Comment:
This may need to include looking at `poll.fm`, but I've never seen any PollDaddy URLs look like that: https://github.com/WordPress/wordpress-develop/blob/c6326694380722ccbdd78e21a2bd0d079383291c/src/wp-includes/class-oembed.php#L69

Reply:
While the risk is minimal, this could throw a PHP warning if the url passed doesn't have an host.